vtex-apps/return-app
# V3 get request list endpoint

This PR brings the endpoint to get a request list. It also fixes an issue preventing the orders to be returned to load when impersonating a user (file: `node/clients/oms.ts`).

### How to test it:

Make a GET request to:

https://rmav3getrequest--powerplanet.myvtex.com/_v/return-request/

The search params available are:
- _page `integer`
- _status `enum`
- _sequenceNumber `string`
- _id `string`
- _createdIn `string` e.g: _createdIn=2022-06-12,2022-07-13
- _orderId `string`
- _userEmail `string`

By default, the requests will only have a summary of the request. If you want to get all the fields for the requests, you can pass another search param: 
- _allFields `string` (any truthy value)

Passing APPKEY and APPTOKEN or VtexIdclientAutCookie you should be able to get any request. If you pass the value of VtexIdclientAutCookie_{store} as VtexIdclientAutCookie you should be able to get only the requests for your user.
